main.go

```diff
@@ -20,16 +20,20 @@ import (
 	"os/exec"
 	"regexp"
 	"runtime"
+	"sync/atomic"
 	"syscall"
 	"time"
 )
 
 var (
-	flagP       = flag.Int("p", runtime.NumCPU(), "run `N` processes in parallel")
-	flagTimeout = flag.Duration("timeout", 10*time.Minute, "timeout each process after `duration`")
-	flagKill    = flag.Bool("kill", true, "kill timed out processes if true, otherwise just print pid (to attach with gdb)")
-	flagFailure = flag.String("failure", "", "fail only if output matches `regexp`")
-	flagIgnore  = flag.String("ignore", "", "ignore failure if output matches `regexp`")
+	flagP             = flag.Int("p", runtime.NumCPU(), "run `N` processes in parallel")
+	flagTimeout       = flag.Duration("timeout", 10*time.Minute, "timeout each process after `duration`")
+	flagKill          = flag.Bool("kill", true, "kill timed out processes if true, otherwise just print pid (to attach with gdb)")
+	flagFailure       = flag.String("failure", "", "fail only if output matches `regexp`")
+	flagIgnore        = flag.String("ignore", "", "ignore failure if output matches `regexp`")
+	flagMaxTime       = flag.Duration("maxtime", 0, "maximum time to run")
+	flagMaxRuns       = flag.Int("maxruns", 0, "maximum number of runs")
+	flagQuitAfterFail = flag.Bool("quitAfterFail", false, "quit after the first detected failure if true")
```

Comment:
Several testing frameworks use `--failfast` for this. Or it could be `--maxfails=1` which would let you collect multiple failures (I don't immediately see a use for that but it would parallel the other `--max` flags)